### PR TITLE
resolves #1338 - adds hreflang metadata

### LINKED ISSUE #1338: Add hreflang metadata
**Is your feature request related to a problem? Please describe.**
We should specify that .co.uk is the alternate url for hreflang=en-gb

**Describe the solution you'd like**
Update the metadata container and component to add the additional information.

```
<link rel="alternate" href="https://www.bbc.co.uk/news/articles/c0000000000o" hreflang="en-gb" />
<link rel="alternate" href="https://www.bbc.com/news/articles/c0000000000o" hreflang="en" />
<link rel="alternate" href="https://www.bbc.com/news/articles/c0000000000o" hreflang="x-default" />
```

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Can add cypress tests, snapshot tests likely sufficient.

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by ChrisBAshton**
Resolves #1338

**Overall change:** Adds three new metadata values. The `.com` variant which has `en`. One is a specific Great Britain version and required the creation of a link which always ends in `.co.uk`.
The x-default is needed for SEO purposes to indicate which hrefLang is the default.

**Code changes:**

- Adds alternateLinks to Metadata component. 
- Adds 3 metadata links to the document head for News. Not necessary for Persian. 
- Updates snapshots

**Testing notes**
- Persian canonical and amp pages **should not** have alternate links: 
- http://localhost:7080/persian/articles/cyddjz5058wo
- http://localhost:7080/persian/articles/cyddjz5058wo.amp
- News canonical and amp pages **should** have alternate links: 
- http://localhost:7080/news/articles/cl55zn0w0l4o
- http://localhost:7080/news/articles/cl55zn0w0l4o.amp

Screen Shot AMP
Screen Shot Canonical

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/components/Metadata/index.jsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ const Metadata = ({
   articleSection,
   brandName,
   canonicalLink,
+  canonicalLinkGb,
```
Comment (dr3) [inline src/app/components/Metadata/index.jsx]: Surely we need to actually pass this from the container?

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/components/Metadata/index.jsx (hunk 1 above)
Comment (ChrisBAshton) [reply src/app/components/Metadata/index.jsx]: 🤦‍♂️

File: src/app/components/Metadata/index.jsx [hunk 2]
```diff
@@ -92,6 +96,7 @@ Metadata.propTypes = {
   articleSection: string,
   brandName: string.isRequired,
   canonicalLink: string.isRequired,
+  canonicalLinkGb: string.isRequired,
```
Comment (sareh) [inline src/app/components/Metadata/index.jsx]: This works for this specific English language page, however it doesn't scale well to other sites.

We should pass an array of alternateLinks to this front-end component, each with a hrefLang and href value. This array should be looped over for this component. The values should be set in the Metadata Container. 

```
const alternateLinks = [
  {
    hrefLang: 'x-default',
    href: 'https://www.bbc.com/news/articles/c0000000001',
  }
  {
    hrefLang: 'en',
    href: 'https://www.bbc.com/news/articles/c0000000001',
  },
  {
    hrefLang: 'en-gb',
    href: 'https://www.bbc.co.uk/news/articles/c0000000001',
  },
];
```

**Review by sareh — CHANGES_REQUESTED** [review]

File: src/app/containers/Metadata/index.jsx [hunk 3]
```diff
@@ -61,6 +62,20 @@ const MetadataContainer = ({ metadata, promo }) => {
                 />
                 <Metadata
                   isAmp={platform === 'amp'}
+                  alternateLinks={[
```
Comment (sareh) [inline src/app/containers/Metadata/index.jsx]: This sets the alternate links to be `en` and `en-gb` for all services. A later PR will be needed to generalise this for all non-English services. This array of values will then need to live in the `ServiceContext`.

**Comment by sareh** [comment]
The codeclimate change suggestions will lead to test files that are less readable by humans, so I suggest disregarding the suggestions for this PR.

**Review by oluoluoxenfree — DISMISSED** [review]

**Review by twinlensreflex — DISMISSED** [review]
Looks good to me 👍 how did you get `rel=alternate` links in the same place in the end?

**Review by sareh — DISMISSED** [review]

**Review by pjlee11 — DISMISSED** [review]
LGTM 👍

**Review by dr3 — DISMISSED** [review]

**Review by pjlee11 — DISMISSED** [review]

**Review by pjlee11 — APPROVED** [review]

**Review by twinlensreflex — APPROVED** [review]

**Review by oluoluoxenfree — APPROVED** [review]
LGTM :)

**Pull request merged**